src/relations/cluster_peers.py

```diff
@@ -0,0 +1,45 @@
+"""NATS Cluster Endpoint handler."""
+import ipaddress
+
+from ops.framework import Object
+
+
+class NatsCluster(Object):
+    """Peer relation `nats-cluster` interface for the NATS charm."""
+
+    def __init__(self, charm, relation_name, listen_on_all_addresses):
+        super().__init__(charm, relation_name)
+        self._relation_name = relation_name
+        if listen_on_all_addresses:
+            # This will create a listening socket for all IPv4 and IPv6 addresses.
+            self._listen_address = ipaddress.ip_address("0.0.0.0")
+            self._ingress_address = ipaddress.ip_address("0.0.0.0")
```

Comment:
Shouldn't this `ipaddress.ip_address("::")` to listen for both IPv4 and IPv6?
As-is, I think this only listens on all IPv4 addresses. WDYT?